supervisor/options.py

```diff
@@ -1481,13 +1481,17 @@ def _exit(self, code):
     def setumask(self, mask):
         os.umask(mask)
 
-    def get_path(self):
+    def get_path(self, env=None):
         """Return a list corresponding to $PATH, or a default."""
-        path = ["/bin", "/usr/bin", "/usr/local/bin"]
+        path = set(["/bin", "/usr/bin", "/usr/local/bin"])
+
         if "PATH" in os.environ:
             p = os.environ["PATH"]
             if p:
-                path = p.split(os.pathsep)
+                path = set(p.split(os.pathsep))
+
+        if env is not None and "PATH" in env:
+            path |= set(env["PATH"].split(os.pathsep))
```

Comment:
These unrelated changes from `list` to `set` need to be reverted.  A `set` is unordered so it is not appropriate for storing `PATH`.  It should remain a `list` as it was.